### ISSUE TITLE
Is there an easy way to exclude Maven test directories from checking?

### ISSUE THREAD

**Issue by noelo-cohelo** [checkstyle/eclipse-cs#132]

**Comment by noelo-cohelo** [comment]
OK, I found it myself ;)

It's under project configuration (right-click on the project -> Properties -> Checkstyle -> Exclude from checking... -> files from packages: <add your test package here>).

Still think this should be included in FAQ.

**Comment by romani** [comment]
ignoring some Checks in maven test folder by checkstyle config - https://github.com/checkstyle/checkstyle/blob/checkstyle-8.17/config/suppressions.xml#L32

**Comment by noelo-cohelo** [comment]
Thank you. What do you think, is it worth putting on FAQ?

I don't use suppresstions.xml, does it mean I need to create one?

**Comment by romani** [comment]
I think you need to create suppression config.

not sure if it worse to be in FAQ. Suppression is a very big topic.

**Comment by mhagnumdw** [comment]
Before, I don't need to do this setting explicitly.

Today I updated the plugin from version 8.23 to 8.32 and the plugin started scanning `src/test/`.

Eclipse version: 2020-03 (4.15.0)

**Comment by romani** [comment]
@Calixte , do you know anything on this change in behavior ?

**Comment by Calixte** [comment]
No. I don't think we introduced any change that would have this consequence.

**Comment by Bananeweizen** [comment]
If test folders shall be ignored completely, wouldn't it be more useful to configure a [BeforeExecutionExclusionFileFilter](https://checkstyle.org/config_filefilters.html#BeforeExecutionExclusionFileFilter)? My understanding was that this one completely avoids matching files from being processed, while all the suppression filters only throw away found issues _after_ processing. Performance wise that might make a big difference on large code bases, wouldn't it?

**Comment by mhagnumdw** [comment]
> Before, I don't need to do this setting explicitly.
> 
> Today I updated the plugin from version 8.23 to 8.32 and the plugin started scanning `src/test/`.
> 
> Eclipse version: 2020-03 (4.15.0)

## After

- Eclipse Checkstyle version 8.23
- `src/test` not scanned, everything is ok here

![image](https://user-images.githubusercontent.com/17688846/82926486-a2c3d100-9f55-11ea-9b1a-3199ce4c4470.png)

[checkstyle-for-8.23.xml.txt](https://github.com/checkstyle/eclipse-cs/files/4683628/checkstyle-for-8.23.xml.txt)

## Before

- Eclipse Checkstyle version 8.32
- `src/test` scanned, here seems to be a problem

![image](https://user-images.githubusercontent.com/17688846/82926537-bc651880-9f55-11ea-994c-3a660ed2aea6.png)

[checkstyle-for-8.32.xml.txt](https://github.com/checkstyle/eclipse-cs/files/4683624/checkstyle-for-8.32.xml.txt)

**Comment by Bananeweizen** [comment]
The Eclipse Checkstyle plugin doesn't differentiate between main sources and test sources (and never did, as far as I know). So it should have been scanning the test sources also in the old version (and that's what it does in all my installations).
However, the plugin will definitively only scan source folders which are configured as part of the classpath. Does your old version really contain a source folder (that's the upper rectangle in your second screenshot), and not just a folder (that's the lower rectangle in your second screenshot). If those are two different installations, then you want to compare the .classpath files of the selected project.

One more thing to look at would be "project context menu -> properties -> checkstyle", there you can specify additional criteria for things to include/exclude in the scan.

**Comment by mhagnumdw** [comment]
> If those are two different installations, then you want to compare the .classpath files of the selected project.

The `.project` file is the same in both. No differences.

The only difference between both projects are:
- the version of the checkstyle plugin for Eclipse
- the `checkstyle.xml` file (because the checkstyle version changes and I needed to make adjustments)

**Comment by JanecekPetr** [comment]
I configured the filter as advised,
```xml
<!-- Ignore test sources -->
<module name="BeforeExecutionExclusionFileFilter">
    <property name="fileNamePattern" value=".*[\\/]src[\\/]test[\\/].*$"/>
</module>
```
and indeed using the Checkstyle Gradle plugin the test sources are excluded. However eclipse still flags the files. Should I file a new ticket as this was originally about something different? Are we misusing it somehow?
